### DISCUSSION TITLE
Try Raspberry Pi Zero W as AirUnit + Raspberry Pi 3B+ as GS -- GS CPU 100%, GPS not used.

### DISCUSSION THREAD

**Discussion by lida2003** [svpcom/wfb-ng#293] [Questions]
I tried Raspberry Pi Zero W (AirUnit) ==> Raspberry Pi 3B+ (GS) with 1280 x 720 @ 30FPS. (with Release 23.01 image)

It seems that Raspberry Pi 3B+ (GS) didn't use GPU for H264 decoding, any thing wrong with gst-launch commands on gs?

### gs command:
```
gst-launch-1.0 udpsrc port=5600 caps='application/x-rtp, media=(string)video, clock-rate=(int)90000, encoding-name=(string)H264' \
               ! rtph264depay ! avdec_h264 ! clockoverlay valignment=bottom ! autovideosink sync=false
```

![图片](https://user-images.githubusercontent.com/1296171/225614191-39530479-2ef4-4a93-b5d9-79bd6b487556.png)

**Comment by svpcom** [comment]
This pipeline is for drone (camera capture), not for GS

**Reply by svpcom** [reply]
For gs see /etc/systemd/system/fpv-video.service

**Reply by lida2003** [reply]
Sorry, I fixed type error.

**Comment by lida2003** [comment]
I hope that the h264 hardware decoder can do the decoding work instead of relying on the CPU.

**Reply by svpcom** [reply]
For RPI3 /etc/systemd/system/fpv-video.service use hw h264 decoder.
For x86 sw decoder is used

**Reply by lida2003** [reply]
OK.

I got an question here. my steps as follows:



1. sudo systemctl start wifibroadcast@gs
2. sudo systemctl start wifibroadcast@drone
3. sudo systemctl start fpv-video
4. drone gst-launch-1.0 720P@45FPS  // OK, I got stable video
5. drone gst-launch-1.0 720P@30FPS  // OK, I got stable video
6. drone gst-launch-1.0 720P@45FPS  // OK, I got stable video
7. drone gst-launch-1.0 800x600@60FPS  // video is gone, just desktop
8. drone gst-launch-1.0 720P@30FPS  // still no video, desktop only
9. sudo systemctl stop fpv-video
10. sudo systemctl start fpv-video // video is back

It seems that I have to stop/start fpv-video when resolution changes.

**Reply by svpcom** [reply]
as I remember (may be wrong) some pi cameras supports only 49fps @720p

**Reply by lida2003** [reply]
If I didn't change the resolution, then it's OK. 
e.g.
720p@30
720P@45
720p@60



if i change from 720P@45 to 800x600@30FPS, the video will stop. (Then I need to stop/start fpv video)
